### PR TITLE
compile_multitarget() should ensure the runtime is built with features common to all targets

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: src/Module.cpp [hunk 1]
```diff
@@ -379,20 +390,33 @@ void compile_multitarget(const std::string &fn_name,
...
     }
 
     // If we haven't specified "no runtime", build a runtime with the base target
     // and add that to the result.
     if (!base_target.has_feature(Target::NoRuntime)) {
-        Target runtime_target = base_target.without_feature(Target::NoRuntime);
+        // Start with a bare Target, set only the features we know are common to all.
+        Target runtime_target(base_target.os, base_target.arch, base_target.bits);
+        // We never want NoRuntime set here.
+        runtime_features_mask &= ~(static_cast<uint64_t>(1) << Target::NoRuntime);
```
Comment (abadams) [inline src/Module.cpp]: We typically use c-style casts instead of static_cast

Comment (steven-johnson) [inline src/Module.cpp]: Torn between multiple coding styles :-)

**Comment by kgnk** [comment]
How about a warning message when a target will never be chosen because of improper ordering?

**Comment by steven-johnson** [comment]
Re: warning message, the gotcha there is that it is dependent on the implementation of halide_can_use_target_features(), which clients can (and do) override; also, not all features have this sort of ordering property.

That said, we could probably special-case for the common case of AVX2-AVX-SSE41 and make a plausible assumption there.

**Pull request merged**